libarchive/archive.h

```diff
@@ -796,6 +796,25 @@ __LA_DECL int archive_write_get_bytes_in_last_block(struct archive *);
 __LA_DECL int archive_write_set_skip_file(struct archive *,
     la_int64_t, la_int64_t);
 
+/*
+ * Set the forced modification time to be applied to all written files.
+ * If clamp is not 0, the given timestamp will only be applied if the file
+ * is newer than it.
+ */
+__LA_DECL int	archive_write_set_forced_mtime(struct archive *,
```

Comment:
I really don't like this approach.   I don't think any changes to libarchive are needed; everything should be handled in bsdtar.